### ISSUE TITLE
warning unused alias

### ISSUE THREAD

**Issue by tcoopman** [zachallaun/mneme#54]
I thought I already saw this mentioned somewhere but can't find anything about it anymore.

The issue is that if you have an alias that's only used in an `auto_assert` block, you get a `warning: unused alias`. But you can't remove the alias, because then the code actually fails.

Elixir: 1.15.2
OTP: 25

**Comment by tcoopman** [comment]
discussion which might be relevant:
https://github.com/elixir-lang/elixir/issues/10115

**Comment by zachallaun** [comment]
Should hopefully be fixed on `main`! If you could confirm, that would be great :)

**Comment by tcoopman** [comment]
@zachallaun awesome work! This fixes the issue!

**Comment by zachallaun** [comment]
Thank you again for the report and for the linked discussion, it was super helpful! The somewhat hacky (but thankfully pretty simple) fix was to extract any aliases used in the assertion and add "dummy usage" for each one, so an assertion that looks like this:

```elixir
auto_assert %MyAlias{} <- some_call()
```

will now expand to something like:

```elixir
_ = MyAlias
... # usual expansion
```

**Comment by zachallaun** [comment]
The fix for both this and #53 should now be in v0.4.3
